lib/pkg.ml

```diff
@@ -242,48 +239,59 @@ let path_of_distrib p =
     | None -> "$(NAME)-$(TAG).tbz"
   in
   dev_repo_is_on_github p >>= fun a ->
-  homepage_is_on_github p >>| fun b ->
-  (if a || b then "releases/download/$(TAG)/" else "releases/") ^ basename
+  homepage_is_on_github p >>= fun b ->
+  let uri =
+    (if a || b then "releases/download/$(TAG)/" else "releases/") ^ basename
+  in
+  name p >>= fun name ->
+  tag p >>= fun tag ->
+  let defs = String.Map.(empty |> add "NAME" name |> add "TAG" tag) in
+  Pat.of_string uri >>| Pat.format defs
 
-let distrib_uri_of_dev_repo p =
-  opam_field_hd p "dev-repo" >>= function
-  | None -> Ok None
-  | Some dev_repo ->
-      let dev_repo =
-        match String.cut ~sep:"git@github.com:" dev_repo with
-        | Some ("", path) -> "https://github.com/" ^ chop_ext path
-        | _ -> (
-            match String.cut ~sep:"git+ssh://git@github.com/" dev_repo with
-            | Some ("", path) -> "https://github.com/" ^ chop_ext path
-            | _ -> chop_git_prefix (chop_ext dev_repo))
-      in
-      path_of_distrib p >>| fun path -> Some (uri_append dev_repo path)
+let uri_of_dev_repo p =
+  opam_field_hd p "dev-repo"
+  >>| Stdext.Option.map ~f:(fun dev_repo ->
```

Comment:
Would it be possible to move all this logic into another module uri.ml, that can just be fed the values of the fields, like `opam_field_hd p "dev-repo" >>| Stdext.Option.map ~f:Uri.of_dev_repo`, and same for other functions, to only keep the opam fields part in this file?

Reply:
That's a good idea! As discussed, I've done that in a [separate PR](https://github.com/ocamllabs/dune-release/pull/349) though since that's not directly related with the bug fix here.